extensions/wikia/Email/EmailController.class.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+namespace Email;
+
+abstract class EmailController extends \WikiaController {
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+
+	/** @var \User The user associated with the current request */
+	protected $currentUser;
+
+	/** @var \User The user this email will be sent to */
+	protected $targetUser;
+
+	protected $hasErrorResponse = false;
+
+	/** @var bool Whether or not to actually send an email */
+	protected $test;
+
+	/**
+	 * Since the children of this class are located in the 'Controller' directory, the default
+	 * location for the template directory would be 'Controller/templates'.  Redefine it to be
+	 * at the root of this extension instead.
+	 *
+	 * @return string
+	 */
+	public static function getTemplateDir() {
+		return dirname( __FILE__ ) . '/templates';
+	}
+
+	public function init() {
+		try {
+			$this->assertCanAccessController();
+
+			$this->currentUser = $this->findUserFromRequest( 'currentUser', $this->wg->User );
+			$this->targetUser = $this->findUserFromRequest( 'targetUser', $this->wg->User );
```

Comment:
What's the reason for having a currentUser and targetUser again? Also, in `assertCanEmail` of this base class we're checking if the current user wants email and is not blocked. It seems like we'd want to be checking that for the target user instead since they're the one receiving the email, right?